### PR TITLE
Use Merkle Tree verification from Trillian

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
Getting rid of legacy Merkle Tree code in this repo.

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
Actually, the verifier interface mismatch is not an issue. After I submit "Part 2" of this, the old interface will be gone so this will explicitly break external code (if any) depending on it, which is better than silently changing method semantics.

**Comment by pavelkalinnikov** [comment]
PTAL.

File: merkletree/tree_hasher.go [hunk 1]
```diff
@@ -14,34 +14,41 @@
...
 
 // HashChildren returns the merkle hash of the two passed in children.
 func (h TreeHasher) HashChildren(left, right []byte) []byte {
-	return h.hasher(append(append([]byte{ct.TreeNodePrefix}, left...), right...))
+	return h.fn(append(append([]byte{ct.TreeNodePrefix}, left...), right...))
+}
+
+// Size is the number of bits in the underlying hash function.
+func (h TreeHasher) Size() int {
+	return len(h.EmptyRoot()) * 8
```
Comment (pavelkalinnikov) [inline merkletree/tree_hasher.go]: This should be the number of bytes instead, the comment in Trillian's `LogHasher` interface was misleading.

Comment (pavelkalinnikov) [reply merkletree/tree_hasher.go]: Done.

**Comment by pavelkalinnikov** [comment]
@daviddrysdale @AlCutter ping

**Comment by pavelkalinnikov** [comment]
Added the "Part 2" to this PR as well. Now it completely removes the legacy Merkle Tree verification code from this repo.

Ping & PTAL.

**Comment by pavelkalinnikov** [comment]
Note: The tests deleted in this PR are now in Trillian (see [#1121](https://github.com/google/trillian/pull/1121)).

File: client/ctclient/ctclient.go [hunk 2]
```diff
@@ -208,12 +209,9 @@ func getInclusionProofForHash(ctx context.Context, logClient client.CheckLogClie
...
 		// If we retrieved an STH we can verify the proof.
-		verifier := merkletree.NewMerkleVerifier(func(data []byte) []byte {
-			hash := sha256.Sum256(data)
-			return hash[:]
-		})
-		if err := verifier.VerifyInclusionProofByHash(rsp.LeafIndex, int64(sth.TreeSize), rsp.AuditPath, sth.SHA256RootHash[:], hash); err != nil {
-			log.Fatalf("Failed to VerifyInclusionProofByHash(%d, %d)=%v", rsp.LeafIndex, sth.TreeSize, err)
+		verifier := merkle.NewLogVerifier(rfc6962.DefaultHasher)
+		if err := verifier.VerifyInclusionProof(rsp.LeafIndex, int64(sth.TreeSize), rsp.AuditPath, sth.SHA256RootHash[:], hash); err != nil {
+			log.Fatalf("Failed to VerifyInclusionProof(%d, %d)=%v", rsp.LeafIndex, sth.TreeSize, err)
```
Comment (AlCutter) [inline client/ctclient/ctclient.go]: These could possibly be `log.Exitf` if we don't think the stack dump will be useful to the user.

Comment (daviddrysdale) [reply client/ctclient/ctclient.go]: No they can't -- base Go lang `log` package has no `Exitf`

**Pull request merged**